addons/mail/static/src/js/systray.js

```diff
@@ -138,14 +123,107 @@ var MessagingMenu = Widget.extend({
     },
     on_click_channel: function (event) {
         var channel_id = $(event.currentTarget).data('channel_id');
-        var channel = chat_manager.get_channel(channel_id);
-        if (channel) {
-            chat_manager.open_channel(channel);
+        if (channel_id == 'channel_inbox') {
+            var message_id = $(event.currentTarget).data('message_id');
+            framework.redirect('mail/view?message_id='+message_id);
+        } else {
+            var channel = chat_manager.get_channel(channel_id);
+            if (channel) {
+                chat_manager.open_channel(channel);
+            }
         }
     },
 });
 
+/**
+ * Menu item appended in the systray part of the navbar, redirects to the next activities of all app
+ */
+var ActivityMenu = Widget.extend({
+    template:'mail.chat.ActivityMenu',
+    events: {
+        "click": "on_activity_menu_click",
+        "click .o_activity_filter_button, .o_mail_channel_preview": "on_activity_filter_click",
+    },
+    start: function () {
+        this.$activities_preview = this.$('.o_mail_navbar_dropdown_channels');
+        chat_manager.bus.on("activity_updated", this, this.update_counter);
+        chat_manager.is_ready.then(this.update_counter.bind(this));
+        this.update_activity_preview();
+        return this._super();
+    },
+    is_open: function () {
+        return this.$el.hasClass('open');
+    },
+    update_counter: function (data) {
```

Comment:
rename this function to _updateCounter, move it to the private section, add docstring